### ISSUE TITLE
Issue plotting 3D tractometry results

### ISSUE THREAD

**Issue by m-petersen** [MIC-DKFZ/TractSeg#109]
Hi Jakob,

first of all thank you for all the effort you are putting into TractSeg.

I am currently performing a groupwise tractometry analysis as implemented in TractSeg. Everything works perfectly fine but plotting the images in 3D -  i.e.,  I am able to extract tractwise metrics and to acquire the lineplots comparing both groups showing sound results. 

However, plotting the significant areas in 3D just works only for every other tract (typically around 5 of the 27 I investigating). Most resulting PNGs are empty black images. I was able to avoid this problem by uncommenting "window.show(renderer, size=img_size, reset_camera=False)" and doing a screenshot.

The other problem is that the 3D plots seem to be flipped horizontally. For instance the lineplots show significantly differing areas in AF_left, but not in AF_right. The 3D plots show corresponding significantly differing areas in AF_right, not in AF_left (please see images below).

I have checked the track files which haven't been touched by me since produced by TractSeg as recommended in your documentation: they are on the right sides.

Do you have an idea where to look to solve these issues?

Thanks in advance.

Line plots
![image](https://user-images.githubusercontent.com/39294105/83798165-1aa3a100-a6a4-11ea-9477-f3827c632b59.png)

3D Plot meant to plot AF_left with significant areas but containing AF_right
![image](https://user-images.githubusercontent.com/39294105/83799190-b5e94600-a6a5-11ea-81d8-d4d81ac86a10.png)

3D Plot meant to plot AF_right without significant areas containing AF_left
![image](https://user-images.githubusercontent.com/39294105/83799522-390a9c00-a6a6-11ea-8683-460453464e20.png)

**Comment by wasserth** [comment]
Are your images in MNI space? If the 3D image is completely black it might be the case that the camera is facing the wrong direction. The camera is adjusted to look towards the right direction for MNI space.

What you could do: 
Uncomment
```
window.show(renderer, size=img_size, reset_camera=False)
print(renderer.get_camera())
```
If you can see the brain now then get the results of `print(renderer.get_camera())` and insert them in `renderer.set_camera(...)` a few lines further up. Maybe you have to do reset_camera=True. Then the camera should be setup the same way as for your screenshot.

Regarding the left/right flip: This could also be a result of your image not having the same left/right orientation as MNI space. Then things can get confused.

I would always recommend to make your subjects have the same orientation (left/right, up/down, front/back) as MNI space. This can be done with `fslreorient2std`.

**Comment by m-petersen** [comment]
Thanks a lot for your prompt response.

I have performed tract segmentation on a group template oriented in accordance with MNI. I figured out a solution by adding a flip of streamline coordinates to plot_bundles_with_metric and changing the affine for the mask actor to beginnings_img.affine:

streamlines = nib.orientations.flip_axis(streamlines, axis=0)
cont_actor = vtk_utils.contour_from_roi_smooth(mask, affine=beginnings_img.affine, color=[.9, .9, .9], opacity=.2, smoothing=50)

I followed you recommendation regarding the 3D tract visualizations. Although I am able to visualize everything with window.show and window.snapshot applying the same camera adjustments to window.record does not help. Since every other tract gets succesfully visualized (the choice of visualized tracts is random for every script execution) I think it's a problem with fury. However, window.snapshot did the trick allowing for automated visualization and saving in png format of the 3D plots.

Cheers,
Marvin
